design/parallel-commit/README.md

```diff
@@ -60,40 +60,50 @@ TODO partial rollback with for_update_ts
 
 
-In order to guarantee SI, commit ts need to satisfy:
-* It is larger than the start ts of all transactions that have not read the lock.
-* It can be larger or smaller than the transaction start ts that are executed at the same time and read the lock.
-* It is smaller than the start ts of subsequent transactions.
+In order to guarantee SI, `commit_ts` of T1 needs to satisfy:
+
+* It is larger than the `start_ts` of any other transaction which has read the old value of a key written by T1.
+* It is smaller than the `start_ts` of any other transaction which reads the new value of any key written by T1.
+
+Note that a transaction executing in parallel with T1 which reads a key written by T1 can have `start_ts` before or after T1's `commit_ts`.
 
 Consider all cases of Parallel Commit:
-* Normal execution process: After Prewrite is completed, it will be returned to the client response. If it is to take the commit ts asynchronously afterwards, it is possible that after the client receives the success, it initiates another transaction to obtain the start ts, and then the commit ts is obtained asynchronously. Not satisfied 3 anymore. The solutions are as follows: after obtaining commit ts and returning success, then commit asynchronously.
-* After Prewrite succeeds, tidb hangs, and the transaction is successfully committed. How to determine commit ts? Of course, you can't get a new one directly from pd, it will not satisfy 3, and also negate the above solution 1, you need to have persistent information to determine the commit ts.
 
-Parallel Commit is considered to be a successful commit after Prewrite is successful. All prewrites form a barrier, and transactions after this must be able to see it, and transactions before this can not see it. How to guarantee this? You need tikv to maintain max start ts. At the same time, prewrite writes to the lock and returns to tidb. Use max(max start ts…) + 1 to submit. When resolve lock, you can recalculate commit ts.
+* Normal execution process: After Prewrite is completed, TiKV will return a response to the client. If it is to take the `commit_ts` asynchronously afterwards, then the client could start a new transaction between the time of receiving the response and the time of TiKV getting a `commit_ts`. The new transaction would therefore read the old value of a key modified by the first transaction which violates RC. A solution is for TiKV to first obtain a timestamp and return that to the client as part of the prewrite response, then use that timestamp to commit.
```

Comment:
>  A solution is for TiKV to first obtain a timestamp and return that to the client as part of the prewrite response, then use that timestamp to commit.

TiDB obtains the commit ts after prewrite succeeds and before returns to the client.